subsys/testsuite/ztest/src/ztest_posix.c

```diff
@@ -0,0 +1,219 @@
+/*
+ * Copyright (c) 2022 Google LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include "cmdline.h" /* native_posix command line options header */
+#include "soc.h"
+#include "tc_util.h"
+#include "ztest_test_new.h"
+
+static const char *test_args;
+static bool list_tests;
+
+static void add_test_filter_option(void)
+{
+	static struct args_struct_t test_filter_s[] = {
+		/*
+		 * Fields:
+		 * manual, mandatory, switch,
+		 * option_name, var_name ,type,
+		 * destination, callback,
+		 * description
+		 */
+		{ false, false, true, "list", NULL, 'b', (void *)&list_tests, NULL,
+		  "List all suite and test cases" },
+		{ false, false, false, "test", "suite::test", 's', (void *)&test_args, NULL,
+		  "Name of tests to run. Comma separated list formatted as "
+		  "\'suiteA::test1,suiteA::test2,suiteB::*\'. An * can be used "
+		  "as a wildcard to run all tests within a suite." },
+		ARG_TABLE_ENDMARKER
+	};
+
+	native_add_command_line_opts(test_filter_s);
+}
+
+NATIVE_TASK(add_test_filter_option, PRE_BOOT_1, 10);
+
+/**
+ * @brief Try to shorten a filename by removing the current directory
+ *
+ * This helps to reduce the very long filenames in assertion failures. It
+ * removes the current directory from the filename and returns the rest.
+ * This makes assertions a lot more readable, and sometimes they fit on one
+ * line.
+ *
+ * Overrides implementation in ztest_new.c
+ *
+ * @param file Filename to check
+ * @returns Shortened filename, or @file if it could not be shortened
+ */
+const char *ztest_relative_filename(const char *file)
+{
+	const char *cwd;
+	char buf[200];
+
+	cwd = getcwd(buf, sizeof(buf));
+	if (cwd && strlen(file) > strlen(cwd) &&
+	    !strncmp(file, cwd, strlen(cwd)))
```

Comment:
Use braces even where not explicitly required: https://docs.zephyrproject.org/3.0.0/contribute/index.html#coding-style